packages/worker/src/index.ts

```diff
@@ -5,60 +5,126 @@ import {
 	parseTranscriptJobMessage,
 	isFailure,
 	deleteMessage,
+	changeMessageVisibility,
+	TranscriptionConfig,
+	TranscriptionOutput,
 } from '@guardian/transcription-service-common';
 import { getSNSClient, publishTranscriptionOutput } from './sns';
+import {
+	getTranscriptionText,
+	convertToWav,
+	createContainer,
+} from './transcribe';
+import {
+	getFile,
+	getS3Client,
+} from '@guardian/transcription-service-common/src/s3';
+import path from 'path';
 
 const main = async () => {
-	const config = await getConfig();
-	const client = getClient(config.aws.region, config.aws.localstackEndpoint);
-
-	// to simulate a transcription job, delay 5 seconds in DEV, 2 minutes in PROD before deleting the message
-	const dummyDelay = config.app.stage === 'DEV' ? 5 : 120;
-	// override timeout to allow enough time for our dummy delay before message becomes available
-	const timeoutOverride = dummyDelay + 1;
-	const message = await getNextMessage(
-		client,
-		config.app.taskQueueUrl,
-		timeoutOverride,
-	);
+	try {
```

Comment:
non-blocking: maybe as we test it might be worth moving to doing more specific error handling with try / catches around each step (get a message, retrieve file, convert to wav, etc.)
We might want to handle errors differently based on the specific exception and which step we're on (e.g. if there's a network error while fetching the file we could retry, but if while converting to wav we find the audio file is malformed, we might want to give up and send a failure notification)

Reply:
Yes definetely, currently every step has its own try catch but it's just logging the error and re-throwing it, so that it'll get caught by the outermost try/catch. When we do more testing and recognise that in some of the steps we need to do differently when an error happens, we can then add actions other than just logging.